parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -102,6 +103,13 @@
 
   public static String PARQUET_READ_PARALLELISM = "parquet.metadata.read.parallelism";
 
+  // configure if we want to use Hadoop's V2 read(bytebuffer) API.
+  // If true, we try to read using the new Hadoop read(ByteBuffer) api. This reads data into the provided
+  // byteBuffer and allows us to potentially take advantage zero-copy read path in Hadoop.
+  // Else, we skip and either read into the byteBuffer's array (if its present) or allocate one and copy
+  // into it.
+  public static String PARQUET_HADOOP_BYTEBUFFER_READ = "parquet.read.use.byte.buffer";
```

Comment:
I'm currently testing something out on this. Basically, we should get the class for ByteBufferReadable, then use it like this:

```
if (bbReadableClass != null && newV2ReaderCtor != null && bbReadableClass.isInstance(stream)) {
  return newV2ReaderCtor.newInstance(stream);
} else {
  return new V1Reader(stream);
}
```